### PR TITLE
feat: added metrics

### PR DESCRIPTION
**Pull request by oopcode**
* Added Prometheus metrics and Prometheus + Graphana examples;
* N.B.: Changed default ports for components.

### CODE REVIEW

File: cmd/hub/main.go [hunk 1]
```diff
@@ -69,8 +67,15 @@ func main() {
 		h.Close()
 	}()
 
+	go func() {
```
Comment (3Hren) [inline cmd/hub/main.go]: I'd rather not to overload the main function by decomposing into a separate one. Moreover the same code is in several places.

File: cmd/locator/main.go [hunk 2]
```diff
@@ -48,6 +50,14 @@ func main() {
 		log.G(ctx).Error("cannot start Locator service", zap.Error(err))
 		os.Exit(1)
 	}
+
+	go func() {
```
Comment (3Hren) [inline cmd/locator/main.go]: Look above.

File: cmd/miner/main.go [hunk 3]
```diff
@@ -71,6 +73,13 @@ func main() {
 		m.Close()
 	}()
 
+	go func() {
```
Comment (3Hren) [inline cmd/miner/main.go]: Look above.

File: cmd/node/main.go [hunk 4]
```diff
@@ -49,6 +51,13 @@ func main() {
 		os.Exit(1)
 	}
 
+	go func() {
```
Comment (3Hren) [inline cmd/node/main.go]: Look above.

File: hub.yaml [hunk 5]
```diff
@@ -1,11 +1,11 @@
 # An endpoint for miner connections.
 # allowed format is "ip:port" to bind Hub to given IP address
 # or ":port" to bind Hub to all available IPs
-endpoint: ":10002"
+endpoint: ":15011"
 
 # Hub as a gateway settings. Can be omitted indicating that the Hub should not
 # be a gateway.
-gateway:
+#gateway:
```
Comment (3Hren) [inline hub.yaml]: Hey! Return back.

File: insonmnia/hub/metrics.go [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package hub
+
+import "github.com/prometheus/client_golang/prometheus"
+
+var (
+	tasksGauge = prometheus.NewGauge(prometheus.GaugeOpts{
```
Comment (3Hren) [inline insonmnia/hub/metrics.go]: Why gauges, not counters?

File: task.yaml [hunk 7]
```diff
@@ -6,7 +6,7 @@ task:
     # Add ability to pull container back
     commit_on_stop: true
     # image name to start on miner, required param
-    name: image:v1
+    name: httpd:latest
```
Comment (3Hren) [inline task.yaml]: ?

File: util/xgrpc/server.go [hunk 8]
```diff
@@ -7,14 +7,17 @@ import (
...
 func NewServer(logger *zap.Logger, extraOpts ...ServerOption) *grpc.Server {
-	opts := newOptions(logger, extraOpts...)
-	srv := grpc.NewServer(
-		append([]grpc.ServerOption{
-			grpc.RPCCompressor(grpc.NewGZIPCompressor()),
-			grpc.RPCDecompressor(grpc.NewGZIPDecompressor()),
-			grpc.UnaryInterceptor(grpc_middleware.ChainUnaryServer(opts.interceptors.u...)),
-			grpc.StreamInterceptor(grpc_middleware.ChainStreamServer(opts.interceptors.s...)),
-		}, opts.options...)...,
+	var (
```
Comment (3Hren) [inline util/xgrpc/server.go]: What is the secret knowledge replacing `:=` by `var`?

Comment (oopcode) [reply util/xgrpc/server.go]: It’s aesthetic!

File: insonmnia/hub/metrics.go (hunk 6 above)
Comment (oopcode) [reply insonmnia/hub/metrics.go]: Counters have only Inc() and no Dec() : )

File: task.yaml (hunk 7 above)
Comment (oopcode) [reply task.yaml]: You have to change image name to something meaningful every time you test tasks. Somebody had to put an end to this. I think @sshaman1101 is with me here.

File: hub.yaml (hunk 5 above)
Comment (oopcode) [reply hub.yaml]: Sure, my bad!

File: insonmnia/hub/metrics.go (hunk 6 above)
Comment (3Hren) [reply insonmnia/hub/metrics.go]: Oh, okay.

File: util/xgrpc/server.go (hunk 8 above)
Comment (3Hren) [reply util/xgrpc/server.go]: I think it's crap! Let's avoid doing unnecessary changes.

**Pull request merged**